manifests/pam.pp

```diff
@@ -1,17 +1,21 @@
-# Class: pam_access::pam
+# @summary Private class - Not to be used directly
 #
-# Private class - Not to be used directly
 # See README.md for usage information
 #
-# [Remember: No empty lines between comments and class definition]
 class pam_access::pam {
 
-  case $::osfamily {
+  $_os_ver = Integer.new( $facts['os']['release']['major'] )
+
+  case $facts['os']['family'] {
     'RedHat': {
-      anchor { 'pam_access::pam::begin': } -> class { 'pam_access::pam::redhat': } -> anchor { 'pam_access::pam::end': }
+      if $_os_ver <= 7 {
+        contain pam_access::pam::redhat7
+      } else {
+        contain pam_access::pam::redhat
+      }
     }
     'Debian': {
-      anchor { 'pam_access::pam::begin': } -> class { 'pam_access::pam::debian': } -> anchor { 'pam_access::pam::end': }
+      contain pam_access::pam::debian
```

Comment:
Why was `contain` used here, rather than just an `include`?

Reply:
I don't recall the exact reason (wish I would have put a comment there), but using `include` here caused timing issues with the pam_access file.

Note the definition at https://puppet.com/docs/puppet/7/lang_containment.html :

> Contained resources are not applied before the container begins, and they finish before the container finishes.

Taking the whole puppet catalog into mind, puppet resolved all dependencies and added any explicit ordering as necessary. Using an `contain` here means the resources will be instantiated at the same time this class is instantiated. In other words, it ensures those "contained" resources will be realized (including any commands to run) during the same time this class runs. Otherwise (with an `include`), the "included" resources could be realized at any point in time, before, during, or after the other resources in this class.